modules/osm/way.js

```diff
@@ -138,31 +138,53 @@ Object.assign(osmWay.prototype, {
     },
 
 
-    isOneWay: function() {
-        // explicit oneway tag..
-        var values = {
-            'yes': true,
-            '1': true,
-            '-1': true,
-            'reversible': true,
-            'alternating': true,
-            'no': false,
-            '0': false
-        };
-        if (values[this.tags.oneway] !== undefined) {
-            return values[this.tags.oneway];
+    /** @returns {boolean} for example, if `oneway=yes` */
+    isOneWayForwards() {
+        if (this.tags.oneway === 'no' || this.tags.oneway === '0') return false;
+
+        for (const key in this.tags) {
+            const value = this.tags[key];
+            const match = osmOneWayForwardTags[key]?.[value];
+            if (match) {
+                return match;
+            }
+        }
+        return false;
+    },
+
+    /** @returns {boolean} for example, if `oneway=-1` */
+    isOneWayBackwards() {
+        if (this.tags.oneway === 'no' || this.tags.oneway === '0') return false;
+
+        for (const key in this.tags) {
+            const value = this.tags[key];
+            const match = osmOneWayBackwardTags[key]?.[value];
+            if (match) {
+                return match;
+            }
         }
+        return false;
+    },
 
-        // implied oneway tag..
-        for (var key in this.tags) {
-            if (key in osmOneWayTags &&
-                (this.tags[key] in osmOneWayTags[key])) {
-                return true;
+    /** @returns {boolean} for example, if `oneway=alternating` */
+    isBiDirectional() {
+        if (this.tags.oneway === 'no' || this.tags.oneway === '0') return false;
+
+        for (const key in this.tags) {
+            const value = this.tags[key];
+            const match = osmBiDirectionalTags[key]?.[value];
+            if (match) {
+                return match;
             }
         }
         return false;
     },
 
+    /** @returns {boolean} */
+    isOneWay() {
+        return this.isOneWayForwards() || this.isOneWayBackwards() || this.isBiDirectional();
+    },
```

Comment:
as this is the most often called function, we could save a few CPU cycles by also storing a merged version of all "oneway" tags in a private variable here. E.g. like this (using lodash's [`merge`](https://lodash.com/docs/4.17.15#merge)):

```js
const osmOneWayTags = merge(osmOneWayForwardTags, osmOneWayBackwardTags, osmBiDirectionalTags);

…

    /** @returns {boolean} */
    isOneWay() {
        if (this.tags.oneway === 'no' || this.tags.oneway === '0') return false;
        for (const key in this.tags) {
            …
        }
        return false;
    },
```